packages/components/src/toast/toast-container.tsx

```diff
@@ -14,43 +14,51 @@ type ToastRootProps = Partial<Pick<RootProps, 'duration' | 'type'>>
 type ToastState = {
   toast: (ToastProps & { rootProps?: ToastRootProps }) | null
   dismiss: () => void
-  positive: (args: {
-    message: string
-    actionProps?: Pick<ToastProps, 'action' | 'onAction'>
-    rootProps?: ToastRootProps
-  }) => void
-  negative: (args: {
-    message: string
-    actionProps?: Pick<ToastProps, 'action' | 'onAction'>
-    rootProps?: ToastRootProps
-  }) => void
-  custom: (args: {
-    message: string
-    icon: React.ReactElement
-    actionProps?: Pick<ToastProps, 'action' | 'onAction'>
-    rootProps?: ToastRootProps
-  }) => void
+  positive: (
+    message: string,
+    options?: {
+      actionProps?: Pick<ToastProps, 'action' | 'onAction'>
+      rootProps?: ToastRootProps
+    }
```

Comment:
```ts
type Options = ToastRootProps & Pick<ToastProps, 'action' | 'onAction'>
```

Reply:
isn't it clearer that way? wouldn't destructuring for `rootProps` instead of its individual fields ensure it's not passed further down where not expected?

Reply:
From a component consumer view, I don't see why they'd need to understand what `root` and `action` are.

The toast API 
```ts
toast.success("Yaaayy!!", {
  duration: 3000,
  action: "Undo",
  onAction: () => void
})
```

is much cleaner than nesting it under some arbitrary fields.

Reply:
which option goes where is an implementational detail.

Reply:
https://github.com/status-im/status-web/pull/517/commits/4e2ed219d1d23ed70749779e875777fbd3ec2a45